DAS/das/utils/logger.py

```diff
@@ -0,0 +1,235 @@
+from pathlib import Path
+import paho.mqtt.client as mqtt
+import paho.mqtt.publish as publish
+import csv
+import time
+import os
+import asyncio
+
+
+CsvConfig = {
+    "delimiter": ',',
+    "quotechar": '`',
+    "quoting": csv.QUOTE_ALL,
+    "skipinitialspace": True,
+    "fieldnames": ["time_delta", "mqtt_topic", "message"],
+}
+
+
+class Record:
+    """Record incoming MQTT messages on selected topics.
+
+    Parameters
+    ----------
+    csv_folder_path : str
+        Filepath of the folder where the logs are to be stored
+    topics : List(str)
+        A list containing the topic strings that are to be subscribed to
+    broker_address : str
+        The IP address that the MQTT broker lives on
+    verbose : bool
+        Specifies whether the incoming MQTT data and warnings are printed
+
+    Attributes
+    ----------
+    TOPICS: List(str)
+        List of topic names
+    _VERBOSE: bool
+        Specifies whether the incoming MQTT data and warnings are printed
+    _START_TIME: `time`
+        The current time used to produce time deltas
+    _LOG_FILE: `File`
+        Open log file object that can be written to
+    _LOG_FILE_WRITER: `csv.DictWriter`
+        Csv writter object that is used to write the data to the _LOG_FILE file
+    _client: `paho.mqtt.client`
+        MQTT client that connects to the broker and recives the messages
+    """
+
+    def __init__(self, csv_folder_path: str, topics: list = ["#"], broker_address: str = "localhost", verbose: bool = False) -> None:
+        # The logger object can subscribe to many topics (if none are selected then it will subscrive to all)
+        self.TOPICS = topics
+        self._VERBOSE = verbose
+        self._START_TIME = time.monotonic()
+
+        # Create csv_folder_path folder if none exists
+        Path(csv_folder_path).mkdir(parents=True, exist_ok=True)
+
+        # Name the csv log file xxxx_log.csv where xxxx is a number
+        previous_log_num = 0
+        for filename in os.listdir(csv_folder_path):
```

Comment:
Might be better to abstract this out into a private method so the constructor looks cleaner.